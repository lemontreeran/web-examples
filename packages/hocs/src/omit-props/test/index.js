import React from 'react'
import { mount } from 'enzyme'

import omitProps from '../src/'

const Target = () => null

describe('omitProps', () => {
  it('should omit multiple props passed in as arguments', () => {
    const EnhancedTarget = omitProps('a', 'b')(Target)
    const wrapper = mount(
      <EnhancedTarget a={1} b={2} c={3} />
    )

    expect(wrapper.find(Target)).toMatchSnapshot()
  })

  it('should do nothing with props if nothing has been passed in', () => {
    const EnhancedTarget = omitProps()(Target)
    const wrapper = mount(
      <EnhancedTarget a={1} b={2} c={3} />
    )

    expect(wrapper.find(Target)).toMatchSnapshot()
  })

  describe('display name', () => {
    const origNodeEnv = process.env.NODE_ENV

    afterAll(() => {
      process.env.NODE_ENV = origNodeEnv
    })

    it('should wrap display name in non-production env', () => {
      process.env.NODE_ENV = 'test'

      const EnhancedTarget = omitProps()(Target)
      const wrapper = mount(
        <EnhancedTarget />
      )

      expect(wrapper).toMatchSnapshot()
    })

    it('should not wrap display name in production env', () => {
      process.env.NODE_ENV = 'production'

      const EnhancedTarget = omitProps()(Target)
      const wrapper = mount(
        <EnhancedTarget />
      )

      expect(wrapper).toMatchSnapshot()
    })
  })
})
