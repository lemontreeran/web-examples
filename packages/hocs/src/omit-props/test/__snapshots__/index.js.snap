// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`omitProps display name should not wrap display name in production env 1`] = `
<OmitProps>
  <Target />
</OmitProps>
`;

exports[`omitProps display name should wrap display name in non-production env 1`] = `
<omitProps(Target)>
  <Target />
</omitProps(Target)>
`;

exports[`omitProps should do nothing with props if nothing has been passed in 1`] = `
<Target
  a={1}
  b={2}
  c={3}
/>
`;

exports[`omitProps should omit multiple props passed in as arguments 1`] = `
<Target
  c={3}
/>
`;
